### PR TITLE
Complement to Rust code in the Chapter array and linked list / Time Complexity.

### PR DESCRIPTION
**Pull request by ElaBosak233**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: docs/chapter_array_and_linkedlist/linked_list.md [hunk 1]
```diff
@@ -369,7 +376,27 @@
 === "Rust"
 
     ```rust title="linked_list.rs"
+    use std::rc::Rc;
+    use std::cell::RefCell;
+
+    #[derive(Debug)]
+    struct ListNode {
```
Comment (krahets) [inline docs/chapter_array_and_linkedlist/linked_list.md:383]: Remove this class. Please refer to other languages.

Comment (ElaBosak233) [reply docs/chapter_array_and_linkedlist/linked_list.md]: Done. Please check. :)

**Comment by krahets** [comment]
Hi @night-cruise , would you like to review this PR?

File: docs/chapter_array_and_linkedlist/array.md [hunk 2]
```diff
@@ -103,7 +103,9 @@
 === "Rust"
 
     ```rust title="array.rs"
-
+    // 初始化数组
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/array.md:106]: /* 初始化数组 */

File: docs/chapter_array_and_linkedlist/linked_list.md [hunk 3]
```diff
@@ -166,7 +166,14 @@
 === "Rust"
 
     ```rust title=""
-
+    use std::rc::Rc;
+    use std::cell::RefCell;
+    // 链表节点类
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/linked_list.md:171]: /* 链表节点类 */

File: docs/chapter_array_and_linkedlist/linked_list.md [hunk 4]
```diff
@@ -166,7 +166,14 @@
...
+    #[derive(Debug)]
+    struct ListNode {
+        val: i32,
+        next: Option<Rc<RefCell<ListNode>>>,
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/linked_list.md:175]: comment

File: docs/chapter_array_and_linkedlist/linked_list.md [hunk 5]
```diff
@@ -369,7 +376,18 @@
 === "Rust"
 
     ```rust title="linked_list.rs"
+    // 初始化链表节点
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/linked_list.md:379]: /* 初始化链表 1 -> 3 -> 2 -> 5 -> 4 */
// 初始化各个节点

File: docs/chapter_array_and_linkedlist/linked_list.md [hunk 6]
```diff
@@ -862,7 +880,25 @@
 === "Rust"
 
     ```rust title=""
+    use std::rc::Rc;
+    use std::cell::RefCell;
 
+    #[derive(Debug)]
+    struct ListNode {
+        val: i32,
+        next: Option<Rc<RefCell<ListNode>>>,
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/linked_list.md:889]: comment

File: docs/chapter_array_and_linkedlist/list.md [hunk 7]
```diff
@@ -119,7 +119,11 @@
 === "Rust"
 
     ```rust title="list.rs"
-
+    // 初始化列表
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/list.md:122]: /* 初始化列表 */

File: docs/chapter_array_and_linkedlist/list.md [hunk 8]
```diff
@@ -233,7 +237,10 @@
 === "Rust"
 
     ```rust title="list.rs"
-
+    // 访问元素
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/list.md:240]: /* 访问元素 */

File: docs/chapter_array_and_linkedlist/list.md [hunk 9]
```diff
@@ -233,7 +237,10 @@
...
+    let num: i32 = list[1];    // 访问索引 1 处的元素
+    // 更新元素
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/list.md:242]: /* 更新元素 */

File: docs/chapter_array_and_linkedlist/list.md [hunk 10]
```diff
@@ -447,7 +454,21 @@
 === "Rust"
 
     ```rust title="list.rs"
+    // 清空列表
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/list.md:457]: /* 清空列表 */

File: docs/chapter_array_and_linkedlist/list.md [hunk 11]
```diff
@@ -447,7 +454,21 @@
...
+    list.clear();
 
+    // 尾部添加元素
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/list.md:460]: /* 尾部添加元素 */

File: docs/chapter_array_and_linkedlist/list.md [hunk 12]
```diff
@@ -447,7 +454,21 @@
...
+    list.push(1);
+    list.push(3);
+    list.push(2);
+    list.push(5);
+    list.push(4);
+
+    // 中间插入元素
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/list.md:467]: /* 中间插入元素 */

File: docs/chapter_array_and_linkedlist/list.md [hunk 13]
```diff
@@ -447,7 +454,21 @@
...
+    list.insert(3, 6);  // 在索引 3 处插入数字 6
+
+    // 删除元素
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/list.md:470]: /* 删除元素 */

File: docs/chapter_array_and_linkedlist/list.md [hunk 14]
```diff
@@ -620,7 +641,17 @@
 === "Rust"
 
     ```rust title="list.rs"
+    // 通过索引遍历列表
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/list.md:644]: /* 通过索引遍历列表 */

File: docs/chapter_array_and_linkedlist/list.md [hunk 15]
```diff
@@ -620,7 +641,17 @@
...
+    let mut count = 0;
+    for (index, value) in list.iter().enumerate() {
+        count += 1;
+    }
 
+    // 直接遍历列表元素
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/list.md:650]: /* 直接遍历列表元素 */

File: docs/chapter_array_and_linkedlist/list.md [hunk 16]
```diff
@@ -717,7 +748,9 @@
 === "Rust"
 
     ```rust title="list.rs"
-
+    // 拼接两个列表
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/list.md:751]: /* 拼接两个列表 */

File: docs/chapter_array_and_linkedlist/list.md [hunk 17]
```diff
@@ -801,7 +834,8 @@
 === "Rust"
 
     ```rust title="list.rs"
-
+    // 排序列表
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/list.md:837]: /* 排序列表 */

File: docs/chapter_computational_complexity/time_complexity.md [hunk 18]
```diff
@@ -171,7 +171,15 @@ $$
 === "Rust"
 
     ```rust title=""
-
+    fn algorithm(n: i32) {
+        let mut a = 2;      // 1 ns
+        a = a + 1;          // 1 ns
+        a = a * 2;          // 10 ns
+        // 循环 n 次
+        for _ in 0..n {     // 1 ns
```
Comment (night-cruise) [inline docs/chapter_computational_complexity/time_complexity.md:179]: // 1 ns ，每轮都要执行 i++

File: docs/chapter_computational_complexity/time_complexity.md [hunk 19]
```diff
@@ -571,7 +594,16 @@ $$
...
 
     ```rust title=""
+    fn algorithm(n: i32) {
+        let mut a = 1;   // +1
+        a = a + 1;      // +1
+        a = a * 2;      // +1
 
+        // 循环 n 次
+        for _ in 0..n { // +1
+            println!("{}", 0); // +1
```
Comment (night-cruise) [inline docs/chapter_computational_complexity/time_complexity.md:604]: +1（每轮都执行 i ++）

**Comment by krahets** [comment]
Hi @ElaBosak233 , Please address the comments from @night-cruise . Thx!

**Comment by ElaBosak233** [comment]
Hi @krahets, I have done it, please check, thank you~

**Comment by krahets** [comment]
@night-cruise Please recheck the PR thx!

File: docs/chapter_array_and_linkedlist/linked_list.md (hunk 4 above)
Comment (night-cruise) [reply docs/chapter_array_and_linkedlist/linked_list.md:175]: @ElaBosak233 Comments are needed here:
```rust
     /* 链表节点类 */
    #[derive(Debug)]
    struct ListNode {
        val: i32,  // 节点值
        next: Option<Rc<RefCell<ListNode>>>,  // 指向下一节点的指针（引用）
    }
```

File: docs/chapter_array_and_linkedlist/linked_list.md (hunk 6 above)
Comment (night-cruise) [reply docs/chapter_array_and_linkedlist/linked_list.md:890]: @ElaBosak233 Also here:
```rust
    /* 双向链表节点类型 */
    #[derive(Debug)]
    struct ListNode {
        val: i32,   // 节点值
        next: Option<Rc<RefCell<ListNode>>>,  // 指向后继节点的指针（引用）
        prev: Option<Rc<RefCell<ListNode>>>,  // 指向前驱节点的指针（引用）
    }
```

File: docs/chapter_array_and_linkedlist/linked_list.md [hunk 20]
```diff
@@ -862,7 +881,25 @@
...
+        prev: Option<Rc<RefCell<ListNode>>>,
+    }
+
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/linked_list.md:893]: And here: /* 构造函数 */

File: docs/chapter_array_and_linkedlist/list.md (hunk 10 above)
Comment (night-cruise) [reply docs/chapter_array_and_linkedlist/list.md:457]: @ElaBosak233 Here should use /**/ comment.

File: docs/chapter_array_and_linkedlist/list.md (hunk 12 above)
Comment (night-cruise) [reply docs/chapter_array_and_linkedlist/list.md:467]: @ElaBosak233 Also here.

File: docs/chapter_array_and_linkedlist/list.md (hunk 13 above)
Comment (night-cruise) [reply docs/chapter_array_and_linkedlist/list.md:470]: @ElaBosak233 And here.

File: docs/chapter_array_and_linkedlist/list.md [hunk 21]
```diff
@@ -801,7 +834,8 @@
 === "Rust"
 
     ```rust title="list.rs"
-
+    /* 排序列表 */
+    list.sort();
```
Comment (night-cruise) [inline docs/chapter_array_and_linkedlist/list.md:838]: Add comment: // 排序后，列表元素从小到大排列

File: docs/chapter_computational_complexity/time_complexity.md [hunk 22]
```diff
@@ -171,7 +171,15 @@ $$
 === "Rust"
 
     ```rust title=""
-
+    fn algorithm(n: i32) {
```
Comment (night-cruise) [inline docs/chapter_computational_complexity/time_complexity.md:174]: Add comment: `// 在某运行平台下`

**Comment by ElaBosak233** [comment]
@night-cruise Done. Please check. Thx.

**Comment by night-cruise** [comment]
> @night-cruise Done. Please check. Thx.

Please see https://github.com/krahets/hello-algo/pull/657#discussion_r1285300982, https://github.com/krahets/hello-algo/pull/657#discussion_r1285301711, https://github.com/krahets/hello-algo/pull/657#discussion_r1285301908 and update the code.

**Comment by ElaBosak233** [comment]
@night-cruise OK.

**Review by night-cruise — APPROVED** [review]
Approve.

**Comment by krahets** [comment]
@night-cruise Thanks! Could you please add my wechat `krahets-jyd`

**Review by krahets — APPROVED** [review]
Thank @ElaBosak233 for the code! Look forward to your next PR.

**Pull request merged**